src/components/Composer.vue

```diff
@@ -314,6 +314,8 @@ export default {
 	},
 	mounted() {
 		this.$refs.toLabel.$el.focus()
+		// event is triggered when user clicks 'new message' in navigation
+		this.$root.$on('newMessage', () => this.reset())
```

Comment:
will it re-register this event handler every time a composer component is rendered? Should there be an `$off` or similar in `unmount`?

Reply:
what happens to the current data in the composer? I guess we should try to save a last draft before clearing the user input, no?

Reply:
> will it re-register this event handler every time a composer component is rendered? Should there be an `$off` or similar in `unmount`?

Good catch, will add this.

>what happens to the current data in the composer? I guess we should try to save a last draft before clearing the user input, no?

I think this is a good question, which maybe should be even discussed in a separate issues (#2510 does not apply here, as there is no draft to save any more after the message have been sent). If the user clicks "New Message" while currently composing another message, does he want to:

1. Save the current message as a draft and create a new, separate message? In this case we not only need to save the last message as a draft (as you suggested) but also make sure to set a new message / draft uid to prevent overwriting the new message.

2. Completely discard the current message (incl. draft versions) and just start over? This is - more or less - what is currently done in this PR.

IMHO, 1) is the better one, so I changed the behavior in the most recent commit (3600208d). 

Note that this made some additional changes necessary to get the `reset()` right (The reset of the text input didn't work correctly as there was no real 2-way data binding between `Composer` and `TextEditor`. This wasn't a problem before, as reset was only used after the messages was sent, so the `TextEditor` would get re-mounted nevertheless)

Reply:
yes 1) makes most sense